utilities/unittests/test_console.py

```
@@ -397,3 +397,43 @@ def test_console_eof_sampler_no_sample(self, mock_get_dir, mock_timeout_sampler)
 
         # Should not change child when no valid sample is found
         assert console.child == original_child
+
+    @patch("utilities.console.pexpect")
+    @patch("utilities.console.get_data_collector_base_directory")
+    def test_console_connect_exception_closes_child(self, mock_get_dir, mock_pexpect):
+        """Test that connect closes child and re-raises exception on _connect failure"""
+        mock_get_dir.return_value = "/tmp/data"
```

Ruff (0.14.14)

Probable insecure usage of temporary file or directory: "/tmp/data"

(S108)